plugin/nfs/nfs.py

```diff
@@ -0,0 +1,496 @@
+            for line in efile:
+                newexp = NFSPlugin._parse_export(shlex.split(line, '#'))
+                if newexp is None:
+                    continue
+                for idx, oldexp in enumerate(exports):
+                    if NFSPlugin._match_export(newexp, oldexp):
+                        exports[idx] = NFSPlugin._merge_exports(oldexp, newexp)
+                        newexp = None
+                if newexp is not None:
+                    exports.append(newexp)
+        return exports
+
+    @staticmethod
+    def _filter_export(exports, export_id):
+        """Remove an entry by its export id"""
+        output = []
+        for exp in exports:
+            if export_id != exp.id:
+                output.append(exp)
+        return output
+
+    @staticmethod
+    def _write_exports(exports, filename=_EXPORTS):
+        """write out the exports list to a file"""
+        try:
+            efile = open(filename, "w")
+        except OSError:
+            raise LsmError(ErrorNumber.PLUGIN_BUG,
+                           'error opening exports for writing:')
+        try:
+            efile.write('# NFS exports managed by libstoragemgmt.'
+                        ' do not edit.\n')
+
+            for exp in exports:
+                common_opts = {}
+                if exp._options is not None:
+                    common_opts = NFSPlugin._parse_options(exp._options)
+
+                if exp._anonuid is not None and exp._anonuid != NfsExport.ANON_UID_GID_NA:
+                    common_opts['anonuid'] = exp._anonuid
+
+                if exp.anongid is not None and exp._anongid != NfsExport.ANON_UID_GID_NA:
+                    common_opts['anongid'] = exp._anongid
+
+                if exp._auth is not None:
+                    common_opts['sec'] = str(exp._auth)
+
+                if exp._export_path is not None:
+                    if not os.path.isdir(exp._export_path):
+                        raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                                       'Export path does not exist')
+
+                if exp._fs_id is not None and exp._export_path is not None:
+                    export_id = get_fsid(exp._export_path)
+                    if export_id != exp._fs_id:
+                        raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                                       'FS ID and Path\'s FS ID do not match')
+
+                hostlist = exp._rw + exp._ro
+                for host in exp._rw:
+                    opts = copy.copy(common_opts)
+                    if host in exp._root:
+                        if 'root_squash' in opts:
+                            del opts['root_squash']
+                        opts['no_root_squash'] = 'no_root_squash'
+                    if 'ro' in opts:
+                        del opts['ro']
+                    opts['rw'] = 'rw'
+                    efile.write("%s %s(%s)\n" % (exp._export_path, host, NFSPlugin._print_option(opts)))
```

Comment:
Please use `pycodestyle` `pep8` to check on `nfs.py` code for PEP8 issues and fix them.